assembly/src/lib.rs

```diff
@@ -17,11 +17,11 @@ use vm_core::{
 mod library;
 pub use library::{Library, LibraryNamespace, LibraryPath, MaslLibrary, Module, Version};
 
-mod procedures;
+pub mod procedures;
 use procedures::{CallSet, Procedure};
 pub use procedures::{ProcedureId, ProcedureName};
```

Comment:
We already export `ProcedureId` and `ProcedureName` publicly - what other structs/functions do we need to export? (if it is just one or two more, maybe it makes sense just to export them rather than the whole module?)

Reply:
For `parsers`, `ModuleAst`, `ProcedureAst`, `ProgramAst` are already exported publicly. Do you need just `Node` and `Instruction` - or something else as well?

Reply:
I need `ModuleAst`, `ProcedureAst`, `ProgramAst`, `Node` and `Instruction`, and as far as I can tell that's all of them.

The compiler need to be able to construct an AST, so if the assembly parser is able to create a struct, then the compiler almost certainly needs to be able to do it as well. That's why I think all the structs should be exported.

Reply:
If that's all that's needed maybe add `Node` and `Instruction` to the line where we already export `ModuleAst`, `ProcedureAst`, and `ProgramAst` (line 25)? Or if you'd prefer to have them grouped into a dedicated namespace, I'd probably prefer `ast` rather than `parsers` (we don't need to rename `parsers` internally, we can just create a wrapper module in `lib.rs` and re-export required structs from there).

Reply:
Done, I think.